PR: https://github.com/odoo/odoo/pull/

From: ea66192276921be1bafe73255fa77c55f3e10ac7
From: Haresh Shyara
Date: 2017-08-09 08:03:08

Breaking data model changes score: 1, change matches:
-    _inherit = 'sale.config.settings'

Total Changes: 124

[REF] crm,sale: Split the applications into 2 separate apps

Purpose
=======

You install CRM but you don't see CRM icon (except in English, as long as sales is not installed) -> confusing
The dashboard with salesteam is not obvious & not useful for simple users: what are those channel boxes about? why don't I see all my activities?, etc.

Specifications
==============

Split CRM & Sales + land in business views straight away

Default Entry Views
~~~~~~~~~~~~~~~~~~~

* New default view for CRM: Pipeline kanban
* New default view for Sales: Quotation list

Dashboards
~~~~~~~~~~

* Move sales teams dashboard to Reporting: Sales Channels.
* Remove the upper part of the sales team dashboard

Menuitems Structure
~~~~~~~~~~~~~~~~~~~

* CRM menu

Pipeline:
	- Leads (optional)
	- Pipeline (default menu, default filter "My Pipeline")
	- Next Activities -> to remove (now there is an icon in top bar for next activities)
	- Quotations (If sale_management installed)
Customers
Phone Calls (if voip, if not yet removed because there is a BE task to use next activities instead)
Leads Management:
	- Scoring Rules
	- Leads Assignation
	- Team Assignation
Reporting:
	- Leads
	- Scoring page views
	- Opp. Assignement
	- Patnerships
	- Pipeline
	- Activities
	- Phonecalls
	- Sales Channels
Configuration:
	- Settings
	- Sales Channels
	- Activity Types
	- Leads & Opportunities:
		- Lead Tags
		- Lost Reasons
	- Resellers
		- Partner Level
		- Partner Activations
* Sales menu:

Orders:
	- Quotations
	- Orders
	- Customers
Invoicing:
	- Orders to Invoice
	- Orders to Upsell
Catalog:
	- Products
	- Promotion Programs
	- Coupon Programs
Reporting:
	- Sales
	- Sales Channels
	- All channels sales orders
Configuration:
	- Settings
	- Sales Channels
	- Sales Orders:
		- Quotation templates
		- Payment acquirers
		- Delivery Methods

Settings
~~~~~~~~

* Split settings form
* Remove recommended apps section
* Remove Timesheets section
* Remove Inventory Management and move shipping connectors to Shipping section
* Remove Integrations section
	- Move Docsaway option to "Quotations & Orders", right after proforma
	- Autocomplete and asterix: move to CRM section
* Restructure and rename sections this way:
	- Product Catalog
	- Pricing
	- Quotations & Orders
	- Shipping
	- Invoicing
	- eBay (if installed)
* Remove all the "Save this page and come back here" when checking a box that installs a new module

* restructure a bit the CRM settings, with 3 sections:

Pipeline
	- Leads
Contacts
	- Phone Validation
	- Customer Autocomplete
Integrations
	- Google Calendar / Synchronize your calendar with Google Calendar (copy from general settings but don't show message "Save this page and come back here...".
	- Asterisk (VOIP)

================================= pseudo patch: =================================

--- a/addons/crm/models/__init__.py
+++ b/addons/crm/models/__init__.py
@@ -6,6 +6,6 @@ from . import calendar
 from . import crm_lead
 from . import crm_stage
 from . import crm_team
-from . import sale_config_settings
+from . import crm_config_settings
 from . import res_partner
 from . import web_planner

--- a/addons/crm/models/sale_config_settings.py
+++ b/addons/crm/models/crm_config_settings.py
@@ -4,15 +4,19 @@
 from odoo import api, fields, models
 
 
-class SaleConfigSettings(models.TransientModel):
-    _inherit = 'sale.config.settings'
+class CrmConfigSettings(models.TransientModel):
+    _name = 'crm.config.settings'
+    _inherit = 'res.config.settings'
 
+    company_id = fields.Many2one('res.company', string='Company', required=True,
+        default=lambda self: self.env.user.company_id)
     alias_prefix = fields.Char('Default Alias Name for Leads')
     alias_domain = fields.Char('Alias Domain', default=lambda self: self.env["ir.config_parameter"].sudo().get_param("mail.catchall.domain"))
     generate_lead_from_alias = fields.Boolean('Manual Assignation of Emails')
     group_use_lead = fields.Boolean(string="Leads", implied_group='crm.group_use_lead')
     module_crm_phone_validation = fields.Boolean("Phone Validation")
     module_crm_voip = fields.Boolean("Asterisk (VoIP)")
+    module_web_clearbit = fields.Boolean("Customer Autocomplete")
 
     def _find_default_lead_alias_id(self):
         alias = self.env.ref('crm.mail_alias_lead_info', False)
@@ -38,7 +42,7 @@ class SaleConfigSettings(models.TransientModel):
 
     @api.model
     def get_values(self):
-        res = super(SaleConfigSettings, self).get_values()
+        res = super(CrmConfigSettings, self).get_values()
         alias = self._find_default_lead_alias_id()
         res.update(
             alias_prefix=alias.alias_name if alias else False,
@@ -48,7 +52,7 @@ class SaleConfigSettings(models.TransientModel):
 
     @api.multi
     def set_values(self):
-        super(SaleConfigSettings, self).set_values()
+        super(CrmConfigSettings, self).set_values()
         self.env['ir.config_parameter'].sudo().set_param('crm.generate_lead_from_alias', self.generate_lead_from_alias)
         alias = self._find_default_lead_alias_id()
         if alias:
